### ISSUE TITLE
Failed to Start: /tmp noexec

### ISSUE THREAD

**Issue by bjmarr** [marktext/marktext#1164]
### Description
The AppImage fails to load if `noexec` is set on `/tmp`.

- [x] Can you reproduce the issue?

### Steps to reproduce
1. Download latest version of marktext
2. `sudo mount -o remount,noexec /tmp`
3. `./marktext-0.14.0-x86_64.AppImage`

**Expected behavior:**
marktext runs normally.

**Actual behavior:**
marktext does not run normally and produces an error.

**Link to an example:**
```bash
└─╼ ./marktext-0.14.0-x86_64.AppImage 
14154
A JavaScript error occurred in the main process
Uncaught Exception:
Error: /tmp/.org.chromium.Chromium.zMF4TY: failed to map segment from shared object
    at process.module.(anonymous function) [as dlopen] (ELECTRON_ASAR.js:160:31)
    at Object.Module._extensions..node (internal/modules/cjs/loader.js:722:18)
    at Object.module.(anonymous function) [as .node] (ELECTRON_ASAR.js:169:18)
    at Module.load (internal/modules/cjs/loader.js:602:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:541:12)
    at Function.Module._load (internal/modules/cjs/loader.js:533:3)
    at Module.require (internal/modules/cjs/loader.js:640:17)
    at require (internal/modules/cjs/helpers.js:20:18)
    at Object.<anonymous> (/tmp/.mount_markteGb0yKn/resources/app.asar/node_modules/keyboard-layout/lib/keyboard-layout.js:4:31)
    at Object.<anonymous> (/tmp/.mount_markteGb0yKn/resources/app.asar/node_modules/keyboard-layout/lib/keyboard-layout.js:59:3)
```

### Versions
- Mark Text: 14.0
- Operating system: Linux Mint 19 Tara

**Comment by fxha** [comment]
Thanks @bjmarr for reporting this issue. AppImage's (loaded by `appimaged`) can only run from a directory that is mounted with executable permissions. You would need to configure your system that the AppImage is mounted to a directory with executable permissions. Otherwise you could try the binary or Flatpak version.

**Comment by fxha** [comment]
Reopening this issue because this affects all versions on Linux except the Flatpak package. The reason for this is because Electron unpack all native node modules at runtime and try to load (execute) them from the temporary directory. If this directory is mounted as `noexec` the application will crash.

I guess we can workaround by unpack the native modules during build time but `AppImage` might be still affected by this issue.

**Comment by fxha** [comment]
This issue is fixed and should work in v0.15.0.
